README.md

```diff
@@ -195,52 +195,78 @@ Install custom reporters as a development dependency and you can use the `--repo
 nyc report --reporter=<custom-reporter-name>
 ```
 
-## Excluding files
+## Selecting files for coverage
 
-You can tell nyc to exclude specific files and directories by adding
-an `nyc.exclude` array to your `package.json`. Each element of
-the array is a glob pattern indicating which paths should be omitted.
+By default, nyc only collects coverage for source files that are visited
+during a test.  It does this by watching for files that are `require()`'d 
+during the test.  Only source files that are visited during a test will appear 
+in the coverage report and contribute to coverage statistics.  
+
+Nyc will cover all files if the `--all` flag is set.  In this case all files will
+appear in the coverage report and contribute to coverage statistics.
+
+Nyc will only cover files that are located under `cwd`, and then only `*.js` files
+or files with extensions listed in in the `nyc.extension` array.
+
+You can reduce the set of covered files by adding `include` and `exclude` 
+filter arrays to your config.  These allow you to shape the set of covered files 
+by specifying glob patterns that can filter files from  the covered set.  The `exclude` 
+array may also use exclude negated glob patterns, these are specified with 
+a `!` prefix, and can restore sub-paths of excluded paths.
 
 Globs are matched using [minimatch](https://www.npmjs.com/package/minimatch).
```

Comment:
I hope we've switched to micromatch by now?

Reply:
Both `test-exclude` and `glob` use `minimatch` to do globbing

Reply:
We switched from micromatch to minimatch on `test-exclude@5` and `nyc@13`.  We were always using `glob` which as @AndrewFinlay points out uses `minimatch`.  I'm not strictly against switching back to `micromatch` but would object to doing this without switching to a `micromatch` based globber.  Obviously this could only be done as part of a semver-major release due to potential risk of changes to edge cases.

Reply:
Hmm I thought minimatch was deprecated at some point in favour of micromatch? Weren't there some security issues or something?

Reply:
From `micromatch` readme:
> not subject to DoS with brace patterns, like minimatch and multimatch

This appears to be an outdated statement, `npm info minimatch@2` says:
> DEPRECATED ⚠️  - Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue

We use 3.0.4 (as does glob).